libs/portals/admin/ids-admin/src/components/Client/Permissions.tsx

```diff
@@ -1,86 +1,110 @@
-import React, { useState } from 'react'
-import AddPermissions from '../forms/AddPermissions/AddPermissions'
-import { ClientFormTypes } from '../forms/EditApplication/EditApplication.action'
+import {
+  Box,
+  Button,
+  Icon,
+  Table as T,
+  Text,
+  toast,
+} from '@island.is/island-ui/core'
+import React, { useContext, useEffect, useRef, useState } from 'react'
+import {
+  ClientFormTypes,
+  EditApplicationResult,
+  schema,
+} from '../forms/EditApplication/EditApplication.action'
 import { ShadowBox } from '../ShadowBox/ShadowBox'
-
-type Permission = {
-  id: string
-  label: string
-  description: string
-  api: string
-  locked?: boolean
-}
+import { useActionData, useParams } from 'react-router-dom'
+import { ClientContext } from '../../shared/context/ClientContext'
+import { AuthAdminClientAllowedScope } from '@island.is/api/schema'
+import AddPermissions from '../forms/AddPermissions/AddPermissions'
+import { getTranslatedValue } from '@island.is/portals/core'
+import isEqual from 'lodash/isEqual'
 
 interface PermissionsProps {
-  data?: Permission[]
+  allowedScopes?: AuthAdminClientAllowedScope[]
 }
 
-const mockData: NonNullable<PermissionsProps['data']> = [
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance:overview',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin:full',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Skattskýrslur',
-    id: '@skatturinn.is/skattskyrslur',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Skatturinn',
-    locked: true,
-  },
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-]
-
-function Permissions({ data = mockData }: PermissionsProps) {
-  const { formatMessage } = useLocale()
+function Permissions({ allowedScopes }: PermissionsProps) {
+  const { formatMessage, locale } = useLocale()
   const [isModalVisible, setIsModalVisible] = useState(false)
+  const params = useParams()
+  const [permissions, setPermissions] = useState<AuthAdminClientAllowedScope[]>(
+    allowedScopes ?? [],
+  )
+  const actionData = useActionData() as EditApplicationResult<
+    typeof schema[typeof ClientFormTypes.permissions]
+  >
+
+  const actionDataRef = useRef(actionData?.data)
```

Comment:
This is initialied as `undefined` on the initial render as the action hasn't been posted. In the useEffect you should just be able to check if action contains data then clear state. We need to make sure that we can repeat this, i.e. we can remove or add scopes and save and then add some more and save.

Reply:
you are right that this gets initialised as undefined until it has been posted. But the thing is that the submit action for this shares a function with a lot of other submit actions, so lets say I submit a change on the display name, then if I would only check if it has data, but don't check if it was a permission that was submitted or a displayName it would reset the added and removed arrays. The isEqual check may be a bit over the top, I will remove that, my idea there was to prevent a reset if the same data gets posted twice.